include/wil/resource.h

```diff
@@ -3454,6 +3463,12 @@ namespace wil
     template<typename T>
     using unique_virtualalloc_ptr = wistd::unique_ptr<T, virtualalloc_deleter>;
 
+    /** Manages a typed pointer allocated with MapViewOfFile
+    A specialization of wistd::unique_ptr<> that frees via UnmapViewOfFile(p).
+    */
+    template<typename T>

```

Comment:
seem this should default to void as that is the common case (in my impl of this that is what I used).
`template<typename T = void>`